### ISSUE TITLE
downloadAttachmentAction corrupts files on download

### ISSUE THREAD

**Issue by sashbash** [mittwald/typo3_forum#230]
Classes/Controller/PostController.php

function downloadAttachmentAction

should end with
readfile($attachment->getAbsoluteFilename());
die();

Otherwise, the resulting file after the download has HTML source code of the page attached to its end. This will likely cause problems when you try to open such broken binary files.

Note:
Some binary formats do still work even if there is unwanted HTML an the end of the file.
I did not recognize the problem at first, because i was only testing with JPG files and most viewers don't seem to care if there is something added to the end of the file. Trying to download a PDF made me realize that there is something wrong, because the file could not be opened after the download.

This issue is related to issue #227 

Best regards
Sascha

**Comment by Sniks6** [comment]
Perfect @sashbash!

Works well now!

Regards, 
Alex

**Comment by sashbash** [comment]
Found a little drawback.
Since we end with die(); 
The download counter will remain unchanged.

To enforce persistence i've added two lines. See below:

```
public function downloadAttachmentAction($attachment) {
		$attachment->increaseDownloadCount();
		$this->attachmentRepository->update($attachment);
						
		//Enforce persistence, since it will not happen because of die() at the end
		$persistenceManager = $this->objectManager->get("TYPO3\\CMS\\Extbase\\Persistence\\Generic\\PersistenceManager");
		$persistenceManager->persistAll();
						
		header('Content-type: ' . $attachment->getMimeType());
		header("Content-Type: application/download");
		header('Content-Disposition: attachment; filename="' . $attachment->getFilename() . '"');
		readfile($attachment->getAbsoluteFilename());
		die();
}
```
Works fine. If there is a better way, please let me know.

**Comment by kpurrmann** [comment]
Resolved #234
